File: src/test/java/com/szadowsz/gospel/core/data/numeric/IntTestCase.java
package com.szadowsz.gospel.core.data.numeric;

import com.szadowsz.gospel.core.data.Struct;
import com.szadowsz.gospel.core.data.Term;
import com.szadowsz.gospel.core.data.Var;
import com.szadowsz.gospel.util.exception.data.InvalidTermException;
import junit.framework.TestCase;

public class IntTestCase extends TestCase {

    public void testIsAtomic() {
        assertTrue(new Int(0).isAtomic());
    }

    public void testIsAtom() {
        assertFalse(new Int(0).isAtom());
    }

    public void testIsCompound() {
        assertFalse(new Int(0).isCompound());
    }

    public void testEqualsToStruct() {
        Term s = new Struct();
        Term zero = new Int(0);
        assertFalse(zero.equals(s));
    }

    public void testEqualsToVar() throws InvalidTermException {
        Term x = new Var("X");
        Term one = new Int(1);
        assertFalse(one.equals(x));
    }

    public void testEqualsToInt() {
        Int zero = new Int(0);
        Int one = new Int(1);
        assertFalse(zero.equals(one));
        Int anotherZero = new Int(0);
        assertTrue(anotherZero.equals(zero));
    }
}

File: src/test/scala/com/szadowsz/gospel/core/data/numeric/IntTestCase.scala
package com.szadowsz.gospel.core.data.numeric

import com.szadowsz.gospel.core.data.{numeric, Var, Term, Struct}
import org.junit.runner.RunWith
import org.scalatest.{Matchers, FlatSpec}
import org.scalatest.junit.JUnitRunner


@RunWith(classOf[JUnitRunner])
class IntTestCase extends FlatSpec with Matchers {

  "An Integer" should " be atomic" in {
    Number(0).isAtomic should be(true)
  }

  "An Integer" should " not be an atom" in {
    Number(0).isAtom should be(false)
  }

  "An Integer" should " not be a compound" in {
    Number(0).isCompound should be(false)
  }

  "An Integer" should " not equivalent to an empty Struct" in {
    val t1 : Term = Number(0)
    val t2 : Term = new Struct()
    t1 should not be t2
  }

  "An Integer" should " not equivalent to an unbound Var" in {
    val t1 : Term = Number(0)
    val t2 : Term = new Var("X")
    t1 should not be t2
  }

  "An Integer of value 1" should " not equivalent to an Integer of value 0" in {
    Number(0) should not be Number(1)
  }

  "An Integer of value 1" should " be equivalent to an Integer of value 1" in {
    Number(1) should be (Number(1))
  }
}
